src/main/java/pl/vm/javaguild/designpatterns/pattern/creational/factorymethod/Department/FinancialDepartment.java

```diff
@@ -0,0 +1,28 @@
+package pl.vm.javaguild.designpatterns.pattern.creational.factorymethod.Department;
+
+import lombok.RequiredArgsConstructor;
+import pl.vm.javaguild.designpatterns.pattern.creational.factorymethod.Employee.Employee;
+import pl.vm.javaguild.designpatterns.pattern.creational.factorymethod.Department.Enums.Section;
+import pl.vm.javaguild.designpatterns.pattern.creational.factorymethod.Department.Enums.Seniority;
+import pl.vm.javaguild.designpatterns.pattern.creational.factorymethod.Employee.FinanceDepartmentEmployee;
+
+/**
+ * FinancialDepartment will produce financial department employees
+ */
+@RequiredArgsConstructor
+public class FinancialDepartment extends Department {
+
+    private Seniority seniority;
```

Comment:
proponuje dodac modyfikatory final do tych zmiennych - i tak tylko je raz przypisujemy w konstruktorze

Reply:
Dokładnie :) Ewentualnie, co ja stosuje, zamiast klasy, możesz wykorzystać record, który zapewnia taką funkcjonalność _out-of-the-box_